### PR TITLE
[Fix](variables) fix session variable does not take effect immediately when set global variable in follower FE

### PR DESCRIPTION
**Pull request by Cai-Yao**
# Proposed changes

Issue Number: close #xxx

## Problem summary

For the non-master node, the `set global xxx = xxx` will not change the variables of the current session, because of `forward()`  to master node execute.
So the `set global xxx = xxx` needs to be executed again to change the current session after `forward()`.

This PR will change behavior, but I think the old behavior is a bug.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/MasterOpExecutor.java [hunk 1]
```diff
@@ -163,6 +179,24 @@ private void forward() throws Exception {
...
     }
 
+    // The SetStmt needs to be executed in follower again
+    // for the current session after master success
+    private void afterForwardExecute() throws DdlException {
+        if (parsedStmt != null) {
+            if (parsedStmt instanceof SetStmt) {
+                SetExecutor executor = new SetExecutor(ctx, (SetStmt) parsedStmt);
+                try {
+                    executor.setSessionVars();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/MasterOpExecutor.java:189]: I think we can set remove the `global` property in `SetStmt` and than call `executor.execute()` method. No need to create a new method `setSessionVars()`.

File: fe/fe-core/src/main/java/org/apache/doris/qe/SetExecutor.java [hunk 2]
```diff
@@ -59,9 +59,24 @@ private void setVariable(SetVar var) throws DdlException {
         }
     }
 
+    private boolean isSessionVar(SetVar var) {
+        return !(var instanceof SetPassVar
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SetExecutor.java:63]: This is error-prone, if other people at a new subclass of `SetVar`, this method may be missed.
How about add a `SetVarType` or something to specify the stmt's type directly?
And write like `return var.setType() == Type.SET_SESSION_VAR`;

Comment (Cai-Yao) [reply fe/fe-core/src/main/java/org/apache/doris/qe/SetExecutor.java:63]: > This is error-prone, if other people at a new subclass of `SetVar`, this method may be missed. How about add a `SetVarType` or something to specify the stmt's type directly? And write like `return var.setType() == Type.SET_SESSION_VAR`;

Yes, I will modify it.

File: fe/fe-core/src/main/java/org/apache/doris/qe/MasterOpExecutor.java (hunk 1 above)
Comment (Cai-Yao) [reply fe/fe-core/src/main/java/org/apache/doris/qe/MasterOpExecutor.java:189]: > I think we can set remove the `global` property in `SetStmt` and than call `executor.execute()` method. No need to create a new method `setSessionVars()`.

Ok, I see what you mean.

**Comment by yiguolei** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**